modules/videoio/src/backend_plugin.cpp

```diff
@@ -407,20 +503,46 @@ void PluginBackendFactory::loadPlugin()
         try
         {
             Ptr<PluginBackend> pluginBackend = makePtr<PluginBackend>(lib);
-            if (pluginBackend && pluginBackend->plugin_api_)
+            if (!pluginBackend)
+                return;
+            if (pluginBackend->capture_api_)
+            {
+                if (pluginBackend->capture_api_->v0.id != id_)
+                {
+                    CV_LOG_ERROR(NULL, "Video I/O: plugin '" << pluginBackend->capture_api_->api_header.api_description <<
+                                       "': unexpected backend ID: " <<
+                                       pluginBackend->capture_api_->v0.id << " vs " << (int)id_ << " (expected)");
+                    return;
+                }
+            }
+            if (pluginBackend->writer_api_)
+            {
+                if (pluginBackend->writer_api_->v0.id != id_)
+                {
+                    CV_LOG_ERROR(NULL, "Video I/O: plugin '" << pluginBackend->writer_api_->api_header.api_description <<
+                                       "': unexpected backend ID: " <<
+                                       pluginBackend->writer_api_->v0.id << " vs " << (int)id_ << " (expected)");
+                    return;
+                }
+            }
+            if (pluginBackend->plugin_api_)
```

Comment:
Perhaps we could show a warning about old API deprecation.

Reply:
FFmpeg wrapper on Windows still uses old Plugin API.